frame/vesting/src/mock.rs

```diff
@@ -153,7 +153,10 @@ impl ExtBuilder {
 			.assimilate_storage(&mut t)
 			.unwrap();
 		let mut ext = sp_io::TestExternalities::new(t);
-		ext.execute_with(|| System::set_block_number(1));
+		ext.execute_with(|| {
+			System::set_block_number(1);
+			Vesting::do_try_state().unwrap();
+		});
```

Comment:
I think this doesn't really test much because it runs before there's any state set up to test?

I suggest that you 
1. test that the try-state hooks actually will detect the issues that they're supposed to, check out how that's done here: https://github.com/paritytech/substrate/blob/master/frame/bags-list/src/list/tests.rs#L353-L378 (also do this with `tips`)
2. pepper `assert_ok!(vesting::do_try_state());`s into existing tests, like you did with `tips`

Reply:
But i still think it's all the same and there's no need to pepper existing tests, the current setup tests the genesis state.

Reply:
Let me know if I'm missing something, but I think we should test all functionality not just the genesis state? 

Like `bags-list` there should be tests ensuring that the `try-state` hooks actually detect the issues that they're supposed to: https://github.com/paritytech/substrate/blob/master/frame/bags-list/src/list/tests.rs#L353-L378

Reply:
Hello @liamaharon , I have made updates to this. Please review

Reply:
You have not added tests ensuring that the try-state hooks detects the issues they are supposed to, as I suggested.

The vesting try-state hooks are currently broken (see other comment), if you had tests that should have helped you to detect them.